zerver/lib/alert_words.py

```diff
@@ -36,26 +34,31 @@ def get_alert_word_automaton(realm: Realm) -> ahocorasick.Automaton:
     return alert_word_automaton
 
 def user_alert_words(user_profile: UserProfile) -> List[str]:
-    return ujson.loads(user_profile.alert_words)
+    alert_word_list = []
+    for alert in AlertWord.objects.filter(user_profile=user_profile):
+        alert_word_list.append(alert.word)
+    return alert_word_list
 
 def add_user_alert_words(user_profile: UserProfile, alert_words: Iterable[str]) -> List[str]:
     words = user_alert_words(user_profile)
 
     new_words = [w for w in alert_words if w not in words]
-    words.extend(new_words)
 
-    set_user_alert_words(user_profile, words)
+    AlertWord.objects.bulk_create(
+        AlertWord(user_profile=user_profile, word=word, realm=user_profile.realm) for word in new_words)
 
-    return words
+    return words+new_words
 
 def remove_user_alert_words(user_profile: UserProfile, alert_words: Iterable[str]) -> List[str]:
     words = user_alert_words(user_profile)
-    words = [w for w in words if w not in alert_words]
+    delete_words = [w for w in alert_words if w in words]
+    for word in delete_words:
+        w = AlertWord.objects.get(user_profile=user_profile, word=word)
+        w.delete()
 
-    set_user_alert_words(user_profile, words)
-
-    return words
+    return user_alert_words(user_profile)
 
 def set_user_alert_words(user_profile: UserProfile, alert_words: List[str]) -> None:
-    user_profile.alert_words = ujson.dumps(alert_words)
-    user_profile.save(update_fields=['alert_words'])
+    AlertWord.objects.filter(user_profile=user_profile).delete()
+    for word in alert_words:
+        AlertWord.objects.create(realm=user_profile.realm, user_profile=user_profile, word=word)
```

Comment:
So, this function still has database queries in a loop.

That said, I kinda would prefer that this function didn't exist in this file, since it's only used by a code path in the `bugdown` unit tests.  Can you add a preparatory commit as the first commit in this branch removing this function (and its caller, `do_set_alert_words`)?  The tests should just be calling `do_add_alert_words`.